Library/Homebrew/extend/git_repository.rb

```diff
@@ -8,102 +8,84 @@
 # @see Utils::Git
 # @api private
 module GitRepositoryExtension
-  extend T::Sig
```

Comment:
This feels non-ideal (more than above). Again, warrants a comment perhaps? Also wondering if making `Settings` just not use `GitRepositoryExtension` would be better?

Reply:
Yeah this is the biggest change tbh and is only because `HOMEBREW_REPOSITORY` uses it. The rest are indeed very trivial.

Reply:
I could make `HOMEBREW_REPOSITORY` only use it later in the loading cycle - even though messing with constants does seem a little smelly.

Reply:
Yeh, I thought the same and agreed it does seem a little smelly.

Reply:
Alternative is to just ditch `GitRepositoryExtension` for `HOMEBREW_REPOSITORY` and just localise it to where it is needed - which is just a diagnostic and SystemConfig.

Reply:
Yeh, that makes more sense to me 👍🏻